internal/service/vpc/resource_aws_vpc_peering_connection.go

```diff
@@ -0,0 +1,337 @@
+package vpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/aiven/terraform-provider-aiven/internal/schemautil"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var aivenAWSVPCPeeringConnectionSchema = map[string]*schema.Schema{
+	"vpc_id": {
+		ForceNew:    true,
+		Required:    true,
+		Type:        schema.TypeString,
+		Description: schemautil.Complex("The VPC the peering connection belongs to.").ForceNew().Build(),
+	},
+	"aws_account_id": {
+		ForceNew:    true,
+		Required:    true,
+		Type:        schema.TypeString,
+		Description: schemautil.Complex("AWS account ID").ForceNew().Build(),
+	},
+	"aws_vpc_id": {
+		ForceNew:    true,
+		Required:    true,
+		Type:        schema.TypeString,
+		Description: schemautil.Complex("AWS VPC ID").ForceNew().Build(),
+	},
+	"aws_vpc_region": {
+		ForceNew: true,
+		Optional: true,
+		Type:     schema.TypeString,
+		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+			return new == ""
+		},
+		Description: schemautil.Complex("AWS region of the peered VPC (if not in the same region as Aiven VPC).").ForceNew().Build(),
+	},
+	"state": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: "State of the peering connection",
+	},
+	"state_info": {
+		Computed:    true,
+		Type:        schema.TypeMap,
+		Description: "State-specific help or error information",
+	},
+	"aws_vpc_peering_connection_id": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: "AWS VPC peering connection ID",
+	},
+}
+
+func ResourceAWSVPCPeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The AWS VPC Peering Connection resource allows the creation and management of Aiven AWS VPC Peering Connections.",
+		CreateContext: resourceAWSVPCPeeringConnectionCreate,
+		ReadContext:   resourceAWSVPCPeeringConnectionRead,
+		DeleteContext: resourceAWSVPCPeeringConnectionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAWSVPCPeeringConnectionImport,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(2 * time.Minute),
+			Delete: schema.DefaultTimeout(2 * time.Minute),
+		},
+
+		Schema: aivenAWSVPCPeeringConnectionSchema,
+	}
+}
+
+func resourceAWSVPCPeeringConnectionCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var (
+		pc     *aiven.VPCPeeringConnection
+		err    error
+		region *string
+	)
+
+	client := m.(*aiven.Client)
+	projectName, vpcID := schemautil.SplitResourceID2(d.Get("vpc_id").(string))
```

Comment:
Wondering if it's possible to use a special combined type for `vpc_id` to get `projectName, vpcID` pair and check errors right at the schema level?